### PR TITLE
Clean up test cases

### PR DESCRIPTION
**Pull request by maxsam4**
### Please check if the PR fulfills these requirements
- [x] The commit message follows our Submission guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce? 
1) Added helper for catching exceptions in test cases/
2) @satyamakgec Added poly initializer helper
3) Ran prettier to make code more readable.

### Does this PR introduce a breaking change?
NO

### Any Other information:
Work has just been started on it. In conjunction with @satyamakgec 

UPDATE
I think we should merge this branch immediately AFTER code freeze so that the new modules can make use of new helpers.
We'll continue working on adding test cases and improving coverage in a new PR.

### CODE REVIEW

File: contracts/modules/Checkpoint/DividendCheckpoint.sol
Comment (maxsam4) [inline contracts/modules/Checkpoint/DividendCheckpoint.sol:58]: This was done to cover the remaining branch as dividend can not be reclaimed before expiry is over. `require(!dividends[_dividendIndex].reclaimed)` is redundant but maybe required later if a module allows reclaiming of dividends before expiry.

Comment (maxsam4) [reply contracts/modules/Checkpoint/DividendCheckpoint.sol:58]: The comment preview does not show it but I have actually moved the require statement above rather than removing.

File: test/helpers/exceptions.js
Comment (maxsam4) [inline test/helpers/exceptions.js]: I added the reason string because we were printing reason for every fail tx. More I use it, more I think that it's redundant. the content inside it("") is the same as this reason. Should we remove this reason and use proper describes and its?

**Comment by maxsam4** [comment]
Checkpoint modules now have 100% coverage except the 3 require(token.transfer) branches. We'll need to create a mock token that returns false on transfer to cover those.

File: contracts/modules/Checkpoint/DividendCheckpoint.sol
Comment (adamdossa) [reply contracts/modules/Checkpoint/DividendCheckpoint.sol:58]: Makes sense. Since this is a base class let's leave the logic as you suggested even though in the current inheriting classes it is redundant (as you can't reclaim a dividend and still be before the expiry).

File: test/helpers/exceptions.js
Comment (adamdossa) [reply test/helpers/exceptions.js]: @satyamakgec to comment here.

**Review by adamdossa — APPROVED** [review]

**Comment by satyamakgec** [comment]
Not merge this PR. I am working on test cases Improvement.

File: test/helpers/exceptions.js
Comment (satyamakgec) [reply test/helpers/exceptions.js]: Yes, @maxsam4 we can remove the reason string. It is same as `it` statement string.

**Comment by maxsam4** [comment]
Do't worry @satyamakgec . I am not done either. Party has just started :)

**Comment by maxsam4** [comment]
Created a script to replace all current try catch blocks with catchRevert(). A lot of stuff must have broke and will need fixing. Gonna continue work tomorrow.

I have also pushed the script "ex.js" if anyone wanna play around. (to be deleted before merging)

**Comment by maxsam4** [comment]
Instrumenting timed out without deepSkip. Works fine on local system though.
We can do two things
1) Increase wait_time in travis
2) Enable deepSkip and not skip oracles and mocks in coverage. (Skipping them with deepSkip gives error)

I am gonna go with the second method for now.

**Pull request merged**
